articles/tutorials/qft-tutorial.md

```diff
@@ -0,0 +1,624 @@
+
+
+### Applying single-qubit and controlled gates
+
+Next, we apply the gates which comprise the operation itself (see the circuit diagram and explanation for the three-qubit example [here](https://en.wikipedia.org/wiki/Quantum_Fourier_transform#Example)).
+Q# already contains many basic quantum gates as operations in the [`Microsoft.Quantum.Intrinsic`](xref:microsoft.quantum.intrinsic) namespace, and these are no exception. 
+To apply an operation to a specific qubit from a register (i.e. a single `Qubit()` from an array `Qubit[]`) we use standard index notation.
+So, applying the [`H`](xref:microsoft.quantum.intrinsic.h) (Hadamard) to the first qubit of our register `qs` takes the form:
+
+```qsharp
+            H(qs[0]);
+```
+
+Besides applying the `H` (Hadamard) gate to individual qubits, the QFT circuit consists primarily of controlled [`R1`](xref:microsoft.quantum.intrinsic.r1) rotations.
+An `R1(θ, <qubit>)` operation in general leaves the $\ket{0}$ component of the qubit unchanged, while applying a rotation of $e^{i\theta}$ to the $\ket{1}$ component.
+
+
+
+#### Controlled operations
+
+Q# makes it extremely easy to condition the execution of an operation upon one or multiple control qubits.
+In general, we merely preface the call with `Controlled`, and the operation arguments change as:
+
+ `Op(<normal args>)` $\to$ `Controlled Op([<control qubits>], (<normal args>))`.
+
+Note that the control qubits must be provided as an array, even if it is a single qubit.
+So, we call the `R1` gates acting on the first qubit (and controlled by the second/third) as:
+
+```qsharp
+            Controlled R1([qs[1]], (PI()/2.0, qs[0]));
+            Controlled R1([qs[2]], (PI()/4.0, qs[0]));
+```
+
+Note that we use the [`PI()`](xref:microsoft.quantum.math.pi) function from the [`Microsoft.Quantum.Math`](xref:microsoft.quantum.math) namespace to define the rotations in terms of pi radians.
+Additionally, we divide by a `Double` (e.g. `2.0`) because dividing by an integer `2` would throw a type error. 
+
+> [!TIP]
+> `R1(π/2)` and `R1(π/4)` are equivalent to the `S` and `T` operations (also in `Microsoft.Quantum.Intrinsic`).
+
+
+After applying the relevant Hadamards and controlled rotations to the second and third qubits:
+
+```qsharp
+            //second qubit:
+            H(qs[1]);
+            Controlled R1([qs[2]], (PI()/2.0, qs[1]));
+
+            //third qubit:
+            H(qs[2]);
+```
+
+we need only apply a [`SWAP`](xref:microsoft.quantum.intrinsic.swap) gate to complete the circuit:
+
+```qsharp
+            SWAP(qs[2], qs[0]);
+```
+
+This is necessary because the nature of the quantum Fourier transform outputs the qubits in reverse order, so the swaps allow for seamless integration of the subroutine into larger algorithms.
+
+
+#### De-allocate qubits
```

Comment:
`s/De-allocate/Deallocate/`

And also on line 165, and anywhere else it appears.